pkg/util/blackbox/types.go

```diff
@@ -66,11 +70,23 @@ func (c *Config) Start(ctx context.Context) {
 			}
 
 			start := time.Now()
-
 			resp, err := c.Cli.Do(c.Req)
-
 			end := time.Now()
 
+			if c.Icli != nil {
+				if resp != nil {
+					request := appinsights.NewRequestTelemetry(c.Req.Method, c.Req.URL.String(), time.Second, resp.Status)
+					request.Id = os.Getenv("RESOURCEGROUP")
```

Comment:
so all requests will have the same value (cluster resource group) for a given cluster?

Reply:
Yes, its like identifier. So our upgrade PR cluster names are already unique. You can play with data in the app-insights instance in our sub already.